executor/executor.go

```diff
@@ -396,6 +400,25 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 			firstError = cmdErr
 		}
 	}
-
+	terminateSleep(shellBin, sourceDir, true) // kill running sleep $SD_TERMINATION_GRACE_PERIOD_SECS
 	return firstError
 }
+
+// terminate long running sleep process for abort, timeout, n after teardown steps
+func terminateSleep(shellBin, sourceDir string, killAll bool) {
+	var stdout, stderr bytes.Buffer
+	shargs := []string{"-e", "-c"}
+	cmdStr := "pids=$(ps -ef | grep '[s]leep' | awk '{print $2}'); pidcnt=$(echo $pids | wc -w); if [ $pidcnt -gt 1 ]; then kill $(echo $pids | awk '{$NF=\"\"}1'); else echo $pids; fi;"
+	if killAll {
+		cmdStr = "pids=$(ps -ef | grep '[s]leep' | awk '{print $2}'); if [ ! -z $pids ]; then kill $pids; else echo $pids; fi;"
+	}
+	shargs = append(shargs, cmdStr)
+	c := exec.Command(shellBin, shargs...)
+	c.Stdout = &stdout
+	c.Stderr = &stderr
+	c.Dir = sourceDir
+	err := c.Run()
+	if err != nil || strings.TrimSpace(stderr.String()) != "" {
```

Comment:
does this makes sure that this function never errors out and worst case it won't kill sleep, but continue with rest of execution ?

Reply:
yes, it will continue with the rest of execution even if it errors.